test/integration/good/compiler-optimizations/cppO1.expected

```diff
@@ -11292,32 +12265,594 @@ stan::math::profile_map& get_stan_profile_data() {
+                                std::ostream* pstream__ = nullptr) : model_base_crtp(0) {
+    int current_statement__ = 0;
+    using local_scalar_t__ = double ;
+    boost::ecuyer1988 base_rng__ = 
+        stan::services::util::create_rng(random_seed__, 0);
+    (void) base_rng__;  // suppress unused var warning
+    static constexpr const char* function__ = "inline_functions_varmat_model_namespace::inline_functions_varmat_model";
+    (void) function__;  // suppress unused var warning
+    local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
+    (void) DUMMY_VAR__;  // suppress unused var warning
+    try {
+      int pos__;
+      pos__ = 1;
+    } catch (const std::exception& e) {
+      stan::lang::rethrow_located(e, locations_array__[current_statement__]);
+    }
+    num_params_r__ = 5 + 5;
+    
+  }
+  
+  template <bool propto__, bool jacobian__ , typename VecR, typename VecI, 
+  stan::require_vector_like_t<VecR>* = nullptr, 
+  stan::require_vector_like_vt<std::is_integral, VecI>* = nullptr> 
+  inline stan::scalar_type_t<VecR> log_prob_impl(VecR& params_r__,
+                                                 VecI& params_i__,
+                                                 std::ostream* pstream__ = nullptr) const {
+    using T__ = stan::scalar_type_t<VecR>;
+    using local_scalar_t__ = T__;
+    T__ lp__(0.0);
+    stan::math::accumulator<T__> lp_accum__;
+    stan::io::deserializer<local_scalar_t__> in__(params_r__, params_i__);
+    int current_statement__ = 0;
+    local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
+    (void) DUMMY_VAR__;  // suppress unused var warning
+    static constexpr const char* function__ = "inline_functions_varmat_model_namespace::log_prob";
+    (void) function__;  // suppress unused var warning
+    
+    try {
+      stan::conditional_var_value_t<local_scalar_t__, 
+        Eigen::Matrix<local_scalar_t__, -1, 1>> p_single_ret_vec;
+      current_statement__ = 1;
+      p_single_ret_vec = in__.template read<
+                           stan::conditional_var_value_t<local_scalar_t__, 
+                             Eigen::Matrix<local_scalar_t__, -1, 1>>>(5);
+      stan::conditional_var_value_t<local_scalar_t__, 
+        Eigen::Matrix<local_scalar_t__, -1, 1>> p_multi_ret_vec;
+      current_statement__ = 2;
+      p_multi_ret_vec = in__.template read<
+                          stan::conditional_var_value_t<local_scalar_t__, 
+                            Eigen::Matrix<local_scalar_t__, -1, 1>>>(5);
+      stan::conditional_var_value_t<local_scalar_t__, 
+        Eigen::Matrix<local_scalar_t__, -1, 1>> tp_single_ret_vec =
+         stan::conditional_var_value_t<local_scalar_t__, 
+           Eigen::Matrix<local_scalar_t__, -1, 1>>(
+           Eigen::Matrix<double, -1, 1>::Constant(5,
+             std::numeric_limits<double>::quiet_NaN()));
+      Eigen::Matrix<double, -1, 1> inline_single_ret_fun_return_sym7__;
```

Comment:
So, I think the DCE scheme is not actually deleting all non-used objects? Else this would be deleted right?